lib/message_bus/pub_sub/redis.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require_relative '../pub_sub_implementation'
+gem 'redis', '>= 5' # Require redis gem v5+
+require 'redis'
+
+module MessageBus
+  module PubSub
+    class Redis
+      include PubSubImplementation
+
+      # @param url [String]
+      def initialize(url:)
+        @redis = ::Redis.new(url: url)
+      end
+
+      # @param channel [String]
+      # @param data [String]
+      # @return [void]
+      def publish(channel, data)
+        @redis.publish(channel, data)
+      end
+
+      # @param channel [String]
+      # @param timeout [Integer]
+      # @yield [channel, message]
+      # @return [void]
+      def subscribe(channel, timeout: 10, &blk)
```

Comment:
I would extract timeout to ENV like `SUB_TIMEOUT_SEK = 10`

Reply:
It should be done then in the appropriate class which uses this pub/sub implementation. In our case it is an AR adapter. But since no other adapters define it - then there is no point to introduce it.